proto/substrait/algebra.proto

```diff
@@ -178,6 +178,18 @@ message JoinRel {
   substrait.extensions.AdvancedExtension advanced_extension = 10;
 }
 
+// A time-series variant of the multi JOIN relational operator left-join-right(s), which joins on an ordered key using inexact matching
+message AsOfJoinRel {
+  JoinRel join = 1;
+  int64 tolerance = 2;
```

Comment:
Can you elaborate on why you chose `int64` here as opposed to an expression or even a literal?

Reply:
I agree this should be more general. The exact definition needed here is a value from an ordered domain with a distance measure. How would that be captured in the proto?

Reply:
In the recent commit I used `Literal` and deferred the domain checking to the consumer.